extensions/vertx-http/runtime/src/main/java/io/quarkus/vertx/http/runtime/devmode/ConfigViewer.java

```diff
@@ -0,0 +1,77 @@
+package io.quarkus.vertx.http.runtime.devmode;
+
+import java.util.Set;
+import java.util.SortedSet;
+import java.util.TreeSet;
+
+import org.eclipse.microprofile.config.Config;
+import org.eclipse.microprofile.config.spi.ConfigSource;
+import org.jboss.logging.Logger;
+
+import io.vertx.core.json.JsonArray;
+import io.vertx.core.json.JsonObject;
+
+/**
+ * Turns a {@link Config} into a JSON object with all config sources and properties as JSON. The config sources are
+ * sorted descending by ordinal, the properties by name. If no config is defined an empty JSON object is returned.
+ *
+ * <p>
+ * A typical output might look like:
+ * </p>
+ *
+ * <pre>
+ * {
+ *   "sources": [
+ *     {
+ *       "source": "source0",
+ *       "ordinal": 200,
+ *       "properties": {
+ *         "key": "value"
+ *       }
+ *     },
+ *     {
+ *       "source": "source1",
+ *       "ordinal": 100,
+ *       "properties": {
+ *         "key": "value"
+ *       }
+ *     }
+ *   ]
+ * }
+ * </pre>
+ */
+class ConfigViewer {
+
+    private static final Logger LOGGER = Logger.getLogger(ConfigViewer.class.getName());
+
+    JsonObject dump(Config config) {
+        JsonObject json = new JsonObject();
+        if (config != null) {
+            if (config.getConfigSources().iterator().hasNext()) {
+                JsonArray jsonSources = new JsonArray();
+                for (ConfigSource source : config.getConfigSources()) {
+                    JsonObject jsonSource = new JsonObject();
+                    jsonSource.put("source", source.getName())
+                            .put("ordinal", source.getOrdinal());
+                    Set<String> propertyNames = source.getPropertyNames();
+                    if (!propertyNames.isEmpty()) {
+                        SortedSet<String> sortedPropertyNames = new TreeSet<>(propertyNames);
+                        JsonObject jsonProperties = new JsonObject();
+                        for (String propertyName : sortedPropertyNames) {
+                            try {
+                                jsonProperties.put(propertyName, source.getValue(propertyName));
```

Comment:
It looks like this is going to print every config value from every config source (as opposed to printing only the effective values).  I assume that this was your intent!

One concern I have is that this will expand property expressions.  If you don't want that (and I suspect you don't since you're printing all the raw config keys rather than just the effective configuration contents), then the whole thing should be enclosed like this:

```java
boolean old = ExpandingConfigSource.setExpanding(false);
try {
    // ... do all the work here ...
} finally {
    ExpandingConfigSource.setExpanding(old);
}

```

Reply:
Another concern I have is that this might expose secrets/passwords!

Reply:
Thanks @dmlloyd 
Right, I only want to print the effective values. Will wrap the code like you suggested.

I don't have a solution how to hide secrets / passwords.

Reply:
Let me post something to the dev list on this. I have an idea... I think.

Reply:
Actually I'll post an issue for it...

Reply:
I've opened #7442 for this.